File: Projeto_2/projeto_2 - Kaella Sales/src/SeguroCarro.cpp
#include "../include/SeguroCarro.h"
#include <iostream>

// Construtor
SeguroCarro::SeguroCarro(Carro& carro, Cliente& cli, float v, std::string vi) 
  : c(carro), SeguroVeiculo(cli, v, vi) {}

// Sobrescrevendo o cálculo de valor para SeguroCarro
void SeguroCarro::CalcularValor() {
    valor += 500; // Exemplo de cálculo adicional para carros
}


void SeguroCarro::exibirDetalhes(){
    std::cout << "Seguro" << std::endl;
    std::cout << "Tipo: Automovel" << std::endl;
   std::cout << "Veículo: " << c.getMarca() << " " << c.getModelo() << " " << c.getAno() << std::endl;
    std::cout << "Valor do seguro: " << valor << std::endl;
    std::cout << "Vigência: " << vigencia << "\n" << std::endl;
}

File: Projeto_2/Projeto2 -_-/src/seguro_carro.py
from src.seguro_veiculo import SeguroVeiculo
from src.cliente import Cliente
from src.carro import Carro

class SeguroCarro(SeguroVeiculo):
    def __init__(self, cliente: Cliente, carros: list[Carro], valor: float, vigencia: str):
        super().__init__(cliente, valor, vigencia)
        self.carros = carros

    def calcular_valor(self):
        taxa_base = 0.05
        return self.valor * (1 + taxa_base * len(self.carros))